src/github.ts

```diff
@@ -106,41 +106,48 @@ export const createTags = async (
   )
 }
 
+const generateReleaseNotes = async (
+  owner: string,
+  repo: string,
+  appOctokit: Octokit,
+  tagName: string
+): Promise<string> => {
+  try {
+    const response = await appOctokit.request(`POST /repos/${owner}/${repo}/releases/generate-notes`, {
+      owner,
+      repo,
+      tag_name: tagName,
+      target_commitish: 'main',
+      headers: {
+        'X-GitHub-Api-Version': '2022-11-28'
+      }
+    })
+    return response.data.body
+  } catch (e) {
+    throw new Error("Couldn't get release notes")
+  }
+}
+
 export const createGithubRelease = async (
   platform: Platform,
   newVersionName: string,
   newVersionCode: number,
   owner: string,
   repo: string,
-  releaseNotes: string,
-  downloadLinks: string,
-  developmentRelease: boolean,
-  dryRun: boolean,
-  appOctokit: Octokit
+  prerelease: string,
+  appOctokit: Octokit,
+  predefinedReleaseNotes?: string
```

Comment:
I think it would be more robust and better, to use also a boolean here, that needs to be definied, like "shouldUsePredefinedReleaseNotes" or "shouldUseReleaseNotesFromReleaseNotesFile" and if it is true, we use the generateReleaseNotes-function. 
E.g. what happens it we do a release with empfy predefinedReleaseNotes.
It would also be nice to now build it in a way we can later also easily move the logic from integreat (build release notes from the folder structure) here.
So maybe the goal would be,
"shouldUseReleaseNotesFromFile"
"shouldUseReleaseNotesFromFolderStructure"
"shouldUsePredefinedReleaseNotes"
as three booleans which are default false, and need to be set to true. If the last one is set to true, the predefinedReleaseNotes need to be set or we throw an error.
Also this is an API and should be specified in a clear manner.

Reply:
@ztefanie i don't get the meaning/use cases of these three booleans
- Normally we pass a json file with the release note content if i'm right (`shouldUseReleaseNotesFromFile`) ?
- If nothing will be passed we can use `shouldGenerateReleaseNotesFromGithub`
Then i dunno what is meant by folder structure? I expected the content of the folder structure will be merged to a file

Reply:
We have two usecases
- build release notes from the folder structure (like we do it in Integreat)
- build release notes from the changelog file (like we do in entitlementcard)

With my comment I was thinking about making the function more flexible to cover both usecases. Therefore we would have needed more parameters. But it is fine as you renamed the function. if we want to add the integreat usecase later to the toolbelt, we can just introduce a new function.